### PR TITLE
Support specifying KeyVault certificate directly rather than by thumbprint

### PR DESCRIPTION
**Pull request by scottbommarito**
The V3 monitoring job must use another means to find the certificate rather than by the thumbprint. This change will allow me to find the certificate myself and then pass it to my `KeyVaultReader`.

### CODE REVIEW

File: src/NuGet.Services.KeyVault/KeyVaultConfiguration.cs [hunk 1]
```diff
@@ -10,12 +10,14 @@ public class KeyVaultConfiguration
     {
         public string VaultName { get; }
         public string ClientId { get; }
-        public string CertificateThumbprint { get; }
-        public StoreName StoreName { get; }
-        public StoreLocation StoreLocation { get; set; }
-        public bool ValidateCertificate { get; }
+        public X509Certificate2 Certificate { get; }
```
Comment (skofman1) [inline src/NuGet.Services.KeyVault/KeyVaultConfiguration.cs]: A configuration class is not supposed to contain any logic, but to be a simple collection of parameters for the class. Please move FindCertificateByThumbprint back.

Comment (scottbommarito) [inline src/NuGet.Services.KeyVault/KeyVaultConfiguration.cs]: I don't think it makes sense in `KeyVaultReader` anymore, given that the class now takes the certificate from this configuration object. Should I make a static utility class to put this function in instead?

Comment (joelverhagen) [inline src/NuGet.Services.KeyVault/KeyVaultConfiguration.cs]: I think that would be better, yes. `CertificateUtility` or something like that.

Comment (joelverhagen) [inline src/NuGet.Services.KeyVault/KeyVaultConfiguration.cs]: After this change there should only be one ctor whose params are 1:1 with properties.

Comment (scottbommarito) [inline src/NuGet.Services.KeyVault/KeyVaultConfiguration.cs]: done

**Pull request merged**